be/src/exec/pipeline/hdfs_chunk_source.cpp

```diff
@@ -0,0 +1,438 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Limited.
+
+#include "exec/pipeline/hdfs_chunk_source.h"
+
+#include "env/env_hdfs.h"
+#include "env/env_s3.h"
+#include "exec/pipeline/scan_operator.h"
+#include "exec/vectorized/hdfs_scan_node.h"
+#include "exec/vectorized/hdfs_scanner_orc.h"
+#include "exec/vectorized/hdfs_scanner_parquet.h"
+#include "exec/vectorized/hdfs_scanner_text.h"
+#include "exec/workgroup/work_group.h"
+#include "exprs/vectorized/runtime_filter.h"
+#include "gutil/map_util.h"
+#include "runtime/current_thread.h"
+#include "runtime/descriptors.h"
+#include "runtime/exec_env.h"
+#include "storage/vectorized/chunk_helper.h"
+#include "util/hdfs_util.h"
+
+namespace starrocks::pipeline {
+using namespace vectorized;
+
+class OpenLimitAllocator {
```

Comment:
What's the meaning  of this class?

Reply:
It's to control max opened hdfs files. I works in old non-pipeline engine, and I don't think it works in current framework.   But inner class `HdfsSanner` are actually using this class, so I keep it.

Reply:
please rename the class name or and more comments